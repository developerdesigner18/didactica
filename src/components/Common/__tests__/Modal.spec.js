import { createLocalVue, mount } from "@vue/test-utils";

import LxpModal from "../Modal.vue";
import { BButton } from "bootstrap-vue";

// create an extended `Vue` constructor
const localVue = createLocalVue();

localVue.component("lxp-modal", LxpModal);
localVue.component("b-button", BButton);

describe("Modal.vue", () => {
  it("render when created ", async () => {
    let wrapper = await getWrapper();
    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
    wrapper.destroy();
  });
  describe("Props", () => {
    let wrapper;

    afterEach(() => {
      wrapper.destroy();
    });

    it("should hide header and footer when `hideHeader` and `hideFooter` is present", async () => {
      wrapper = await getWrapper();
      await wrapper.setProps({
        hideHeader: true,
        hideFooter: true
      });

      const modalHeader = wrapper.find(".modal-header");
      expect(modalHeader.exists()).toBe(false);

      const modalFooter = wrapper.find(".modal-footer");
      expect(modalFooter.exists()).toBe(false);
    });
  });

  describe("Events", () => {
    let wrapper;

    beforeEach(async () => {
      wrapper = await getWrapper();
    });

    afterEach(() => {
      wrapper.destroy();
    });

    it("should emit `modal-close` when close is clicked", async () => {
      let courseCard = wrapper.find(".lxp-modal__close");
      await courseCard.trigger("click");
      expect(wrapper.emitted("modal-close")).toBeTruthy();
    });
  });
});

function getWrapper() {
  return mount(LxpModal, {
    propsData: {
      title: "Modal Title",
      centered: true,
      show: true
    },
    localVue
  });
}
