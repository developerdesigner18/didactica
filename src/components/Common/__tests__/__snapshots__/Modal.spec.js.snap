// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal.vue render when created  1`] = `
<div
  class="lxp-modal"
>
  <div
    class="modal fade"
  >
    <div
      class="modal-dialog modal-dialog-centered modal-md"
      role="document"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <h5
            class="modal-title"
          >
            Modal Title
          </h5>
           
          <button
            class="lxp-modal__close"
            type="button"
          >
            <img
              alt="close"
              loading="lazy"
              src="https://files.lxp.academy.who.int/didactica/assets/images/modal/close.svg"
            />
          </button>
        </div>
         
        <div
          class="modal-body"
        />
         
        <div
          class="modal-footer"
        >
           
          <div
            class="d-flex"
          >
            <button
              class="btn btn-outline-primary btn-sm lxp-button"
              type="button"
            >
              
              Cancel
            
            </button>
             
            <button
              class="btn btn-primary btn-sm lxp-button"
              type="button"
            >
              Save
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="modal-backdrop"
  />
</div>
`;
